src/libraries/System.Net.Sockets/ref/System.Net.Sockets.cs

```diff
@@ -412,6 +412,8 @@ public partial class Socket : System.IDisposable
         public void SendFile(string? fileName) { }
         public void SendFile(string? fileName, byte[]? preBuffer, byte[]? postBuffer, System.Net.Sockets.TransmitFileOptions flags) { }
         public void SendFile(string? fileName, System.ReadOnlySpan<byte> preBuffer, System.ReadOnlySpan<byte> postBuffer, System.Net.Sockets.TransmitFileOptions flags) { }
+        public System.Threading.Tasks.ValueTask SendFileAsync(string? fileName, System.ReadOnlyMemory<byte> preBuffer, System.ReadOnlyMemory<byte> postBuffer, System.Net.Sockets.TransmitFileOptions flags, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public System.Threading.Tasks.ValueTask SendFileAsync(string? fileName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
@geoffkizer do you know why did nullability change here during the API review (https://github.com/dotnet/runtime/issues/42591#issue-706560540 -> https://github.com/dotnet/runtime/issues/42591#issuecomment-696912247). Does anyone need `socket.SendFile(null)`?

Reply:
> `socket.SendFile(null)`

In this case only the pre- and post-buffer will be sent (if given). Whether anyone needs this or not I can't judge...but don't believe so.

Reply:
I'm not quite sure what is going on here...

Existing SendFile and BeginSendFile both have an overload that just takes `string? fileName`. So presumably we wanted SendFileAsync to match, which makes sense, even though SendFile(null) doesn't seem very useful.

However, the documentation has both this listed as `string fileName`, i.e. not nullable. E.g. here: https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.beginsendfile?view=net-5.0

I would suggest we should be consistent with SendFile and BeginSendFile and allow null here. However we should also follow up on the doc issue, and we may want to end up making all of these non-null.

Reply:
Ah, see here: https://github.com/dotnet/runtime/pull/42535

That explains the doc discrepancy -- that change was post-5.0, while the docs are based on 5.0.

Let's be consistent with https://github.com/dotnet/runtime/pull/42535 here.

Reply:
> So presumably we wanted SendFileAsync to match, which makes sense, even though SendFile(null) doesn't seem very useful.

That makes sense it's clear now, thanks!